### PR TITLE
#25738 Adding API integration

### LINKED ISSUE #25738: Build Logic for Showing Unread Announcements
### Parent Issue

https://github.com/dotCMS/core/issues/25733

### Task

_No response_

### Proposed Objective

Same as Parent Issue

### Proposed Priority

Same as Parent Issue

### Acceptance Criteria

Any time you close the Announcements Menu, that will clear all of the "Unread" messages.

Announcements Menu should close when you click off it, or click on any Announcement, or the "View All" button, or any links from the Announcement Menu.

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by manuelrojas**
### Proposed Changes
- Added a test for the `unreadAnnouncements` computed property in the `AnnouncementsService` using Spectator.
- Modified the `AnnouncementsService` to use `localStorage` for storing and retrieving announcements data.
- Updated the test suite to cover scenarios with both new and no new announcements.
- Modified the service to handle announcements retrieval from `localStorage` in the `unreadAnnouncements` computed property.
- Integrated an API call to fetch announcements using `HttpClient` in the `AnnouncementsService`

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
image

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/api/services/dot-announcements.ts/dot-announcements.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                identifier: 'test-announcement-id',
+                url: 'https://www.example.com'
+            }
+        ];
+
+        localStorage.setItem('announcementsData', JSON.stringify(storedAnnouncements));
+
+        const isNewAnnouncement = spectator.service.unreadAnnouncements();
+
+        expect(isNewAnnouncement).toBeFalsy();
```
Comment (oidacra) [inline core-web/apps/dotcms-ui/src/app/api/services/dot-announcements.ts/dot-announcements.service.spec.ts:85]: use `.toBe(false)` instead `.toBeFalsy()`

File: core-web/apps/dotcms-ui/src/app/view/components/dot-toolbar/components/dot-toolbar-announcements/dot-toolbar-announcements.component.html [hunk 2]
```diff
@@ -1,23 +1,32 @@
...
+                <span class="announcements__label">{{ item.title }}</span>
+                <span class="announcements__date">{{ item.announcementDateAsISO8601 | date }}</span>
+            </div>
+        </a>
     </li>
+    }
 </ul>
 
 <div class="announcements__container">
+    <script type="module" src=""></script>
```
Comment (rjvelazco) [inline core-web/apps/dotcms-ui/src/app/view/components/dot-toolbar/components/dot-toolbar-announcements/dot-toolbar-announcements.component.html:25]: Are we using this `script` tag?

File: core-web/apps/dotcms-ui/src/app/api/services/dot-announcements.ts/dot-announcements.service.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+export class AnnouncementsService {
+    private http = inject(HttpClient);
+    private announcementsUrl = '/api/v1/announcements';
+
+    private announcementsData$ = this.http.get<Announcement[]>(this.announcementsUrl).pipe(
+        pluck('entity'),
+        catchError(() => [])
+    );
+
+    announcements: Signal<Announcement[]> = toSignal(this.announcementsData$, {
```
Comment (rjvelazco) [inline core-web/apps/dotcms-ui/src/app/api/services/dot-announcements.ts/dot-announcements.service.ts:31]: Nice!

**Review by rjvelazco — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/view/components/dot-toolbar/components/dot-toolbar-announcements/dot-toolbar-announcements.component.html (hunk 2 above)
Comment (manuelrojas) [reply core-web/apps/dotcms-ui/src/app/view/components/dot-toolbar/components/dot-toolbar-announcements/dot-toolbar-announcements.component.html:25]: Removed!

File: core-web/apps/dotcms-ui/src/app/api/services/dot-announcements.ts/dot-announcements.service.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        );
+
+        const isNewAnnouncement = newAnnouncementIds.some(
+            (id) => !storedAnnouncementIds.includes(id)
+        );
+
+        return isNewAnnouncement;
+    });
+
+    saveAnnouncementsData: Signal<void> = computed(() => {
```
Comment (KevinDavilaDotCMS) [inline core-web/apps/dotcms-ui/src/app/api/services/dot-announcements.ts/dot-announcements.service.ts:62]: I think a computed needs to return a value (You can no return, but the purpose of the computed is compute a value based on other value), here is used as effect
Maybe we can add an effect on service constructor, or take other strategy to update the localStorage

Comment (zJaaal) [reply core-web/apps/dotcms-ui/src/app/api/services/dot-announcements.ts/dot-announcements.service.ts:62]: Indeed if a computed signal is a void, it should be an effect.

**Review by zJaaal — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/api/services/dot-announcements.ts/dot-announcements.service.ts (hunk 4 above)
Comment (fmontes) [reply core-web/apps/dotcms-ui/src/app/api/services/dot-announcements.ts/dot-announcements.service.ts:62]: TIL!

File: core-web/apps/dotcms-ui/src/app/api/services/dot-announcements.ts/dot-announcements.service.ts
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/api/services/dot-announcements.ts/dot-announcements.service.ts:1]: @dotCMS/frontend We are starting to use Signals, is this a right approach? so now we can have services that have no methods and only properties that are Signals? is this a recommended pattern?

For me, this looks suuuuuper weird.

This service looks like a state management that combine data from API and localstorage and then merge it into a Signal.

Should we start using https://ngrx.io/guide/signals/signal-store

Too many questions but I think is important to start align ourselves into how are we going to use signals.

**Review by fmontes — CHANGES_REQUESTED** [review]

**Review by zJaaal — APPROVED** [review]

**Review by rjvelazco — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/view/components/dot-toolbar/components/dot-toolbar-announcements/store/dot-announcements.store.ts [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+                        url: `${announcement.url}?utm_source=dotcms&utm_medium=application&utm_campaign=announcement_menu`
+                    };
+                });
+
+                this.setState({
+                    announcements: modifiedAnnouncements,
+                    readAnnouncements: this.hasUnreadAnnouncements(announcements)
+                });
+
+                return announcements;
```
Comment (rjvelazco) [inline core-web/apps/dotcms-ui/src/app/view/components/dot-toolbar/components/dot-toolbar-announcements/store/dot-announcements.store.ts:57]: tap doesn't need to return a value

File: core-web/apps/dotcms-ui/src/app/view/components/dot-toolbar/components/dot-toolbar-announcements/store/dot-announcements.store.ts [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+    readonly announcementsSignal: Signal<Announcement[]> = this.selectSignal(
+        (state) => state.announcements
+    );
+
+    readonly readAnnouncements: Signal<boolean> = this.selectSignal(
+        (state) => state.readAnnouncements
+    );
+
+    unreadAnnouncements = this.updater((state) => {
```
Comment (rjvelazco) [inline core-web/apps/dotcms-ui/src/app/view/components/dot-toolbar/components/dot-toolbar-announcements/store/dot-announcements.store.ts:71]: Shouldn't this be `readonly`?

**Review by fmontes — CHANGES_REQUESTED** [review]

**Review by fmontes — APPROVED** [review]

**Pull request merged**